server/src/main/java/org/elasticsearch/repositories/RepositoriesMetrics.java

```diff
@@ -61,4 +65,25 @@ public RepositoriesMetrics(MeterRegistry meterRegistry) {
             )
         );
     }
+
+    /**
+     * Create the map of attributes we expect to see on repository metrics
+     */
+    public static Map<String, Object> createAttributesMap(
+        RepositoryMetadata repositoryMetadata,
+        OperationPurpose purpose,
+        String operation
+    ) {
+        return Map.of(
+            "repo_type",
+            repositoryMetadata.type(),
+            "repo_name",
+            repositoryMetadata.name(),
+            "operation",
+            operation,
+            "purpose",
+            purpose.getKey()
+        );
+    }
```

Comment:
It seemed like it'd be a good thing to define this in one place for consistency between repository implementations?